cEP-0020.md

```diff
@@ -0,0 +1,244 @@
+Levels and Badges. For each of these elements, we will have a set of rules to
+describe the operation of the gamified environment. i.e.:
+
+### Activities
+
+Let's start by defining some activities which could be tracked:
+
+- Create Issues
+  - Labels?
+  - No. of comments?
+- Create Pull Requests
+  - Label?
+  - CI status?
+  - No. of Gitmate errors?
+  - No. and type of Comments?
+- Comments
+  - Type?
+- Gitter Messages
+  - Good?
+  - Bad?
+
+The above activities will be tracked seperately based on their category. i.e.:
+
+- There are five types of Issues and Pull Requests that could be tracked:
+  - type/bug
+  - difficulty/newcomer
+  - difficulty/low
+  - difficulty/medium
+  - difficulty/high
+- While tracking Pull Requests, following factors will also be considered:
+  - CI status
+    - Passed
+    - Failed
+  - Comments
+    - No. of simple comments
+    - No. of review comments
+  - Gitmate Errors:
+    - 1 to 5
+    - 5 to 10
+    - 10 and beyond
+- Following are the types of comments that could be tracked:
+  - Issue comments
+  - Review comments
+
+### Points
+
+Points are a simple tool to reward behavior, and they provide an excellent
+base upon which to build a gamified design. Now lets assign some points
+to each of the tracked activities mentioned above:
+
+- Create Issue
+  - 10 points for difficulty/newcomer
+  - Increase in 10 points based on difficulty labels
+  - Increase in 2 points with the increase in no. of comments
+- Create Pull Requests
+  - If CI passes and There are no Gitmate errors
+    - 15 points for intial and then increase
+  - If CI passes and Gitmate errors are b/w 1 to 5
+    - 10 points for intial and then increase
+  - If CI passes and Gitmate errors are b/w 5 to 10
+    - 5 points for intial and then increase
+  - If CI passes and Gitmate errors are b/w 10 and beyond
+    - No points given
+  - If CI failed
+    - No points given until the CI has passed
+- Made Comments
+  - 2 points per issue comments
+  - 3 points per review comments
+- Gitter Messages
+  - No points for Bad messages
+  - 1 points per Good message
+
+### Levels
+
+To increase a player’s competitive instinct, we can implement a series of
+levels that confer rank as newcomers become more active. A good and
+straightforward way to award levels in a new gamified design is to base them
+on point thresholds. As players earn points, they move up an incremental
+series of levels. To infuse the level with meaning i.e.: a ranking within the
+system, the levels should be named in a way that indicates status.
+
+But the levels wouldn't be enough motivation to go until newcomers get
+some special privileges after completing each level:
+
+- Level-I:
+  - Name: Beginner
+  - Points_required = 50
+  - Persmission to ping maintainers on Gitter
+- Level-II:
+  - Name: Intermideate
+  - Points_required = 150
+  - Permission to use corobo basic functionalities
+    - Invite newcomers to org
```

Comment:
this isnt a basic corobo functionality.  this is current maintainers only.  We may make it developers only, or a really high number of points, but not level II.

look through corobo plugins for other privileges in the rules.

But remove all mention of corobo , and add a note that implementing these privileges is beyond the scope the gamification system.  The gamification system will merely expose a list of privileges that each person has, and other systems are expected to adapt to use these privileges.